slither/solc_parsing/variables/variable_declaration.py

```diff
@@ -200,6 +222,47 @@ def _init_from_declaration(self, var: Dict, init: bool):  # pylint: disable=too-
                 self._variable.initialized = True
                 self._initializedNotParsed = var["children"][1]
 
+    def _get_init_value(self, ty : Type) -> Expression:
+
```

Comment:
I think we need a data location check. When the data location is "storage", the variable is initialized as a null reference, instead of the default value.

```solidity
pragma solidity ^0.8.0;

contract ImplicitInitializer {

    string name;

    function storage_var() public view {
        string storage v2; // v2 is a null reference.
        v2 = name;
    }

}
```

Reply:
There doesn't seem to be any way to represent a null reference in SlithIR. However, the solc compiler will produce an error if we try to use an uninitialized state variable. So if we refrain from generating any code at the declaration site, we can rely on it being initialized later using an assignment.